SmartDeviceLink/SDLIAPSession.m

```diff
@@ -139,6 +273,51 @@ - (SDLStreamErrorHandler)streamErroredHandler {
     };
 }
 
+- (SDLStreamHasSpaceHandler)streamHasSpaceHandler {
+    __weak typeof(self) weakSelf = self;
+    
+    return ^(NSStream *stream) {
+        __strong typeof(weakSelf) strongSelf = weakSelf;
+        
+        if (strongSelf.isDataSession){
+            SessionSendHandler sendCompletion;
+            @synchronized (strongSelf) {
+                sendCompletion = strongSelf.sendBlock;
+            }
+            
+            if (sendCompletion != nil){
+                NSError *err = nil;
+                BOOL sendComplete = sendCompletion(&err);
+                if (err != nil){
+                    NSLog(@"ERROR!!! Output stream reported %@", err);
```

Comment:
Also Alex asked about reproducing the specific issue - iOS terminating the app due to high CPU load in the background because of the race condition between the main thread and the send completion block on the transmit queue. The best way to reproduce is as follows:

1. Launch a background enabled SDL app such as Spotify and connect to the head unit
2. Background the app -- turn off the display as this enables iOS aggressive power management.
3. Disconnect the app from the head unit while a number of buffers are in flight to the head unit, say during a large file upload.
4. Reconnect the app to the head unit. It may be necessary to do several cycles of 3 and 4.

Result: The app will stop playing media and when it is brought to the foreground, it will disappear. The device logs will show that the app has been terminated due to excessive CPU usage in the thread backing the iAP transmit queue.

Based on how frequently you are able to reproduce the issue, I am open to making the PR a hotfix or a standard bug fix. I apologize for being a bit rushed yesterday. Sleep deprivation has a way of doing that.